src/routes/checkout/Address/Address.js

```diff
@@ -1,5 +1,329 @@
-import React from "react";
+import React, { useState, useContext, useEffect, useRef } from "react";
 
+import { onAuthStateChanged } from "firebase/auth";
+import { AppContext } from "../../../context/app-context";
+import { UserContext } from "../../../context/user/user-context";
+import { SET_CHECKOUT_ADDRESS } from "../../../constant";
+
+import { doc, getDoc, setDoc } from "firebase/firestore";
+
+import {
+  ThemeProvider,
+  Container,
+  Typography,
+  Grid,
+  Autocomplete,
+  TextField,
+  Button,
+  CircularProgress,
+} from "@mui/material";
+import { ButtonStyles, responsiveTheme, textFieldStyles } from "./muiStyles";
 export default function Address() {
-  return <div>Address</div>;
+  const { dispatch: userDispatch } = useContext(UserContext);
+  const { auth, db } = useContext(AppContext);
+
+  const [loading, setLoading] = useState(false);
+
+  // Reference values of inputs
+  const buildingRef = useRef();
+  const apartmentNumberRef = useRef();
+  const messageRef = useRef();
+
+  useEffect(() => {
+    onAuthStateChanged(auth, async (user) => {
```

Comment:
You don't need `onAuthStateChanged` here because, once, the user must sign in in order to access this page, and second, we already fetched all user's data when the user signed in. You just need to get the data from the Context